Source/DafnyLanguageServer.Test/Synchronization/DiagnosticsTest.cs

```diff
@@ -48,8 +46,6 @@ decreases y
 }".TrimStart();
       var documentItem = CreateTestDocument(source);
       await client.OpenDocumentAndWaitAsync(documentItem, CancellationToken);
-      var diagnostics = await GetLastDiagnostics(documentItem, CancellationToken);
-      Assert.AreEqual(0, diagnostics.Length);
```

Comment:
> Do you have any motivation for not publishing diagnostics that I am not aware of?

It seems better not to send notifications that have no effect on the client.

Reply:
A client could definitely decide to render a diagnostic window differently depending on whether they are published an empty, or if they are not published.

For example, a client could decide to display an initial placeholder message "waiting for diagnostics" and then removing it once it receives empty diagnostics.

From this (slightly ambiguous) specification:
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/

it says that:
>When a file changes it is the server’s responsibility to re-compute diagnostics and push them to the client

I originally disagreed not publishing diagnostics if they were the same as before, but because it made some tests simpler, I ended up agreeing on that "hack", but I think that the spirit is that it should actually push diagnostics, at least the first time, to notify that it's alive, e.g. when opening a new file.

Reply:
I like that you referenced the LSP spec, but I feel it doesn't answer our question.

Let's look at the practical side. The Microsoft VSCode LSP client won't display anything differently if you send empty diagnostics after opening a file. However, our VSCode extension could have the status bar use the fact that first diagnostics have not arrived to determine whether it's showing "Resolving..." or not. If we do that, then I think we can really ditch the compilationStatus API.

I'll undo this change in this PR, and then we may or may not discuss it in a future PR :-)